src/wikitextprocessor/parser.py

```diff
@@ -541,22 +536,19 @@ def template_parameters(
                                 )
                             continue
 
-                if (is_named and 
-                        (isinstance(parameter_name, int) 
-                        or len(parameter_name)) > 0
-                ):
+                if is_named and len(parameter_name) > 0:  # type: ignore
                     parameters[parameter_name].append(parameter)
                 else:
                     parameters[unnamed_parameter_index].append(parameter)
 
         for p_name, p_value in parameters.items():
             if isinstance(p_value, list) and len(p_value) == 1:
-                final_parameters[p_name] = p_value[0]
+                parameters[p_name] = p_value[0]
             else:
-                final_parameters[p_name] = p_value
+                parameters[p_name] = p_value
 
-        self._template_parameters = final_parameters 
-        return final_parameters
+        self._template_parameters = parameters
+        return parameters
```

Comment:
We are returning defaultdict() here, which is why I changed around this bit of code in the first place, not just type-checking. If some other pieces of code receives a defaultdict() that returns empty values when it should throw an exception instead, that's going to be really annoying to debug. Please read my commit messages and don't just revert everything!

Reply:
Sorry I miss the commit message, I convert it to normal dictionary now.

You see this function already makes you a little confused, it shouldn't be more complicated.